syntax = "proto3";

package proto;

option go_package = "task-service/proto";

import "google/protobuf/timestamp.proto";

service TaskService {
    rpc CreateTask(CreateTaskRequest) returns (CreateTaskResponse);
    rpc UpdateTask(UpdateTaskRequest) returns (UpdateTaskResponse);
    rpc DeleteTask(DeleteTaskRequest) returns (DeleteTaskResponse);
    rpc GetTask(GetTaskRequest) returns (GetTaskResponse);
    rpc ListTasks(ListTasksRequest) returns (ListTasksResponse);
}

message Task {
    string id = 1;
    string user_id = 2;
    string title = 3;
    string description = 4;
    string status = 5;
    google.protobuf.Timestamp created_at = 6;
    google.protobuf.Timestamp updated_at = 7;
}

message CreateTaskRequest {
    string id = 1;
    string user_id = 2;
    string title = 3;
    string description = 4;
}

message CreateTaskResponse {
    Task task = 1;
}

message UpdateTaskRequest {
    string id = 1;
    string user_id = 2;
    string title = 3;
    string description = 4;
    string status = 5;
}

message UpdateTaskResponse {
    Task task = 1;
}

message DeleteTaskRequest {
    string id = 1;
    string user_id = 2;
}

message DeleteTaskResponse {
    bool success = 1;
}

message GetTaskRequest {
    string id = 1;
}

message GetTaskResponse {
    Task task = 1;
}

message ListTasksRequest {
    int32 page = 1;
    int32 page_size = 2;
}

message ListTasksResponse {
    repeated Task tasks = 1;
}
